prac_07/project_management.py

```diff
@@ -0,0 +1,122 @@
+        elif choice == "F":
+            filter_by_date(projects)
+        elif choice == "A":
+            add_project(projects)
+        elif choice == "U":
+            update_project(projects)
+        print(MENU)
+        choice = input(">>> ").upper()
+
+
+def format_date(date):
+    """Return date input to format as dd/mm/yyyy - accepts date as yyyy/mm/dd"""
+    project_start_date = date.split("/")
+    start_date_as_datetime = \
+        datetime.date(int(project_start_date[2]), int(project_start_date[1]),
+                      int(project_start_date[0]))
+    formatted_date = datetime.date.strftime(start_date_as_datetime, "%d/%m/%Y")
+    return formatted_date
+
+
+def load_data_from_file(file, data):
+    """Loads data from project file"""
+    in_file = open(file, 'r')
+    in_file.readline()
+    for line in in_file:
+        parts = line.strip().split('\t')
+        project_start_date = format_date(parts[1])
+        project = Project(parts[0], project_start_date, int(parts[2]),
+                          float(parts[3]), int(parts[4]))
+        if project not in data:
+            data.append(project)
+    in_file.close()
+
+
+def save_data_to_file(file, data):
+    """Prints data to an output file"""
+    out_file = open(file, "w")
+    print("Name	Start Date	Priority	Cost Estimate	Completion Percentage", file=out_file)
+    for project in data:
+        print(f"{project.name}\t{project.start_date}\t{project.priority}\t"
+              f"{project.estimate}\t{project.completion}", file=out_file)
+    out_file.close()
+
+
+def display_projects_by_completion(projects):
+    """Prints projects separated by completed/incomplete"""
+    print("Incomplete projects:")
+    for project in projects:
+        if not project.is_complete():
+            print(f"\t{project}")
+    print("Completed projects:")
+    for project in projects:
+        if project.is_complete():
+            print(f"\t{project}")
+
+
+def filter_by_date(projects):
+    """Shows projects that start after a certain date"""
+    date_input = input("Show projects that start after date (dd/mm/yyyy): ")
+    formatted_date = format_date(date_input)
+    for project in projects:
+        if project.start_date >= formatted_date:
+            print(project)
+
+
+def add_project(projects):
+    """Adds a project to the list"""
+    print("Let's add a new project")
+    name = input("Name: ")
+    start_date = input("Start date (dd/mm/yyyy): ")
+    formatted_date = format_date(start_date)
+    priority = int(input("Priority: "))
+    estimate = float(input("Estimate: "))
+    completion = int(input("Percent Complete: "))
+    project = Project(name, formatted_date, priority, estimate, completion)
+    if project not in projects:
+        projects.append(project)
+
+
+def update_project(projects):
+    """Updates completion or priority of projects"""
+    for project in projects:
+        print(f"{projects.index(project)} {project}")
```

Comment:
Just use `enumerate` for this. You don't need to work backwards to determine what loop number you're up to... this is a for loop, so we already know. 
Ues `index` when you don't know what the index is - like finding something in a list.